File: javascript/8kyu/src/test/n001_n010/n002/even_or_odd/even_or_odd.test.js
import {describe, it} from 'mocha';
import {expect} from 'chai';
import even_or_odd from '../../../../main/n001_n010/n002/even_or_odd/even_or_odd.js';

describe('Function even_or_odd', () => {
  context('With (1)', () => {
    it ('Return Odd', () => {
      expect('Odd').to.equal(even_or_odd(1));
    });
  });

  context('With (2)', () => {
    it ('Return Even', () => {
      expect('Even').to.equal(even_or_odd(2));
    });
  });

  context('With (3)', () => {
    it ('Return Odd', () => {
      expect('Odd').to.equal(even_or_odd(3));
    });
  });

  context('With (4)', () => {
    it ('Return Even', () => {
      expect('Even').to.equal(even_or_odd(4));
    });
  });
});

File: java/8kyu/src/test/java/_8kyu/n001_n010/n002/even_or_odd/EvenOrOddTest.java
package _8kyu.n001_n010.n002.even_or_odd;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Nested;
import org.junit.jupiter.api.Test;

@DisplayName("EvenOrOddClass")
public class EvenOrOddTest {
    @DisplayName("#evenOrOdd")
    @Nested
    class Describe_evenOrOdd {
        @DisplayName("With (1)")
        @Nested
        class Context_with_1 {
            @DisplayName("Return Odd")
            @Test
            void It_return_odd() {
                final String expected = "Odd";
                final String actual = EvenOrOdd.evenOrOdd(1);

                Assertions.assertEquals(expected, actual);
            }
        }

        @DisplayName("With (2)")
        @Nested
        class Context_with_2 {
            @DisplayName("Return Even")
            @Test
            void It_return_odd() {
                final String expected = "Even";
                final String actual = EvenOrOdd.evenOrOdd(2);

                Assertions.assertEquals(expected, actual);
            }
        }

        @DisplayName("With (3)")
        @Nested
        class Context_with_3 {
            @DisplayName("Return Odd")
            @Test
            void It_return_odd() {
                final String expected = "Odd";
                final String actual = EvenOrOdd.evenOrOdd(3);

                Assertions.assertEquals(expected, actual);
            }
        }

        @DisplayName("With (4)")
        @Nested
        class Context_with_4 {
            @DisplayName("Return Even")
            @Test
            void It_return_odd() {
                final String expected = "Even";
                final String actual = EvenOrOdd.evenOrOdd(4);

                Assertions.assertEquals(expected, actual);
            }
        }
    }
}
